### PR TITLE
Add "retrieving returns delivery report email from sap" example

### PR DESCRIPTION
**Pull request by Osara-B**
## Purpose
Add example code for demonstrating the process of retrieving customer returns delivery items from SAP and generating a report grouping items by product and sending an email

## Checklist

- [ ] add example
- [ ] Add README.md file

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Examples
+
+The S/4 HANA Delivery and Transporation Ballerina connectors provide practical examples illustrating usage in various
+scenarios. Explore
+these [examples](https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.delivery/tree/main/examples), covering
+use cases like accessing S/4HANA Customer Returns Delivery API and S/4HANA Inbound Delivery API.
```
Comment (niveathika) [inline examples/README.md]:
```suggestion
use cases like integrating paper delivery note system with S/4HANA and generating weekly reports from S/4HANA system. 
```

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+1. [S/4HANA to Gmail Intergration](https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.delivery/tree/main/examples/returns-from-sap) -
```
Comment (niveathika) [inline examples/README.md]:
```suggestion
1. [Weekly Report Generation](https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.delivery/tree/main/examples/returns-from-sap) -
```

File: examples/returns-from-sap/.devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (niveathika) [inline examples/returns-from-sap/.devcontainer.json]: Unnecessary file

File: examples/returns-from-sap/.gitignore [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Ballerina generates this directory during the compilation of a package.
```
Comment (niveathika) [inline examples/returns-from-sap/.gitignore]: Unnecessary file

File: examples/returns-from-sap/Ballerina.toml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+[package]
+org = "osara_bandara"
```
Comment (niveathika) [inline examples/returns-from-sap/Ballerina.toml]:
```suggestion
org = "wso2"
```

File: examples/returns-from-sap/README.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
+# Send a reminder on approval of pending orders
```
Comment (niveathika) [inline examples/returns-from-sap/README.md]: Heading?

File: examples/returns-from-sap/README.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+This example demonstrates leveraging the `sap.s4hana.api_customer_returns_delivery_srv_0002:Client` in Ballerina for S/4HANA API interactions. It
+specifically showcases how to retrieve customer returns delivery items and send an email containing a report on the returns delivery items
+in the S/4HANA SD module.
+## Prerequisites
```
Comment (niveathika) [inline examples/returns-from-sap/README.md]: Formatting

File: examples/returns-from-sap/README.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+Refer to the [Setup Guide](https://central.ballerina.io/ballerinax/sap/latest#setup-guide) for necessary credentials (
+hostname, username, password).
+
+### 2. Setup the Salesforce Client
+
+Refer to the [Setup Guide](https://central.ballerina.io/ballerinax/salesforce/latest#setup-guide) for necessary
+credentials (client ID, secret, tokens).
+
+### 3. Setup the Salesforce Listener
+
```
Comment (niveathika) [inline examples/returns-from-sap/README.md]: Is there a salesforceintegration?

File: examples/returns-from-sap/README.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+clientId = "<Client_ID>"
+clientSecret = "<Client_Secret>"
+refreshToken = "<Refresh_token>"
+
+[gmailConfig]
+recipientAddress = "<Recipient_Address>"
+senderAddress = "<Sender_Address>"
+environment = "DEVELOPER"
+
+```
```
Comment (niveathika) [inline examples/returns-from-sap/README.md]: ????

File: examples/returns-from-sap/main.bal [hunk 10]
```diff
@@ -0,0 +1,91 @@
+import ballerina/io;
```
Comment (niveathika) [inline examples/returns-from-sap/main.bal]: License

File: examples/returns-from-sap/main.bal [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+configurable string hostname = ?;
+configurable string username = ?;
+configurable string password = ?;
+
+configurable string refreshToken = ?;
+configurable string clientId = ?;
+configurable string clientSecret = ?;
+
+configurable string recipientAddress = ?;
+configurable string fromAddress = ?;
```
Comment (niveathika) [inline examples/returns-from-sap/main.bal]: Shall we group them togeather based on the client?

File: examples/returns-from-sap/main.bal [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+            clientId,
+            clientSecret
+        }
+    }
+);
+
+public function main() returns error? {
+
+    returnDel:Client returnDelClient = check new (config, hostname);
+    io:println("Client is created successfully");
```
Comment (niveathika) [inline examples/returns-from-sap/main.bal]: Debug logs in final PR?

File: examples/returns-from-sap/main.bal [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+    string query = "CreationDate gt datetime'" + getUTC() + "'";
```
Comment (niveathika) [inline examples/returns-from-sap/main.bal]:
```suggestion
    string query = "CreationDate gt datetime'" + getStartOfWeekTimestamp() + "'";
```

File: examples/returns-from-sap/main.bal [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+
+    returnDel:CollectionOfA_ReturnsDeliveryItemWrapper returnsDeliveries = check returnDelClient->listA_ReturnsDeliveryItems(
+        \$filter = query
```
Comment (niveathika) [inline examples/returns-from-sap/main.bal]: Give this inline

File: examples/returns-from-sap/main.bal [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+    foreach var Item in groupedItems {
+        string material = Item.Material ?: "";
+        string deliveryDocStr = Item.DeliveryDocument.toString();
+        string totalQuantity = Item.ActualDeliveryQuantity.toString();
+
+        tableRows += "<tr><td>" + material + "</td>";
+        tableRows += "<td>" + deliveryDocStr + "</td>";
+        tableRows += "<td>" + totalQuantity.toString() + "</td></tr>";
+    }
+    return tableStart + tableRows + tableEnd;
```
Comment (niveathika) [inline examples/returns-from-sap/main.bal]: Can  use queries for this too

**Review by niveathika — CHANGES_REQUESTED** [review]

File: examples/returns-from-sap/main.bal (hunk 15 above)
Comment (Osara-B) [reply examples/returns-from-sap/main.bal]: Updated in [Add reviewed changes and documentation](https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.delivery/pull/15/commits/918d0df8497a939d2833b9ed1b4509b83f1f89c0)

File: examples/returns-from-sap/main.bal (hunk 14 above)
Comment (Osara-B) [reply examples/returns-from-sap/main.bal]: Updated in [Add reviewed changes and documentation](https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.delivery/pull/15/commits/918d0df8497a939d2833b9ed1b4509b83f1f89c0)

File: examples/returns-from-sap/main.bal (hunk 13 above)
Comment (Osara-B) [reply examples/returns-from-sap/main.bal]: Updated in [Add reviewed changes and documentation](https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.delivery/pull/15/commits/918d0df8497a939d2833b9ed1b4509b83f1f89c0)

File: examples/README.md (hunk 1 above)
Comment (Osara-B) [reply examples/README.md]: Updated in [Add reviewed changes and documentation](https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.delivery/pull/15/commits/918d0df8497a939d2833b9ed1b4509b83f1f89c0)

File: examples/returns-from-sap/.devcontainer.json (hunk 3 above)
Comment (Osara-B) [reply examples/returns-from-sap/.devcontainer.json]: Updated in [Add reviewed changes and documentation](https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.delivery/pull/15/commits/918d0df8497a939d2833b9ed1b4509b83f1f89c0)

File: examples/README.md (hunk 2 above)
Comment (Osara-B) [reply examples/README.md]: Updated in [Add reviewed changes and documentation](https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.delivery/pull/15/commits/918d0df8497a939d2833b9ed1b4509b83f1f89c0)

File: examples/returns-from-sap/Ballerina.toml (hunk 5 above)
Comment (Osara-B) [reply examples/returns-from-sap/Ballerina.toml]: Updated in [Add reviewed changes and documentation](https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.delivery/pull/15/commits/918d0df8497a939d2833b9ed1b4509b83f1f89c0)

File: examples/returns-from-sap/main.bal (hunk 11 above)
Comment (Osara-B) [reply examples/returns-from-sap/main.bal]: Updated in [Add reviewed changes and documentation](https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.delivery/pull/15/commits/918d0df8497a939d2833b9ed1b4509b83f1f89c0)

**Review by niveathika — APPROVED** [review]

**Pull request merged**